src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,119 @@
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.jdbc.exception.DataProcessingException;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.model.Manufacturer;
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
+    private static final String INSERT_QUERY = "INSERT INTO manufacturers (name, country)"
+            + " VALUES (?,?);";
+    private static final String GET_QUERY
+
+            = "SELECT * FROM manufacturers WHERE is_deleted = FALSE AND id = ?;";
+    private static final String GET_ALL_QUERY
+            = "SELECT * FROM manufacturers WHERE is_deleted = FALSE;";
+    private static final String UPDATE_QUERY =
+            "UPDATE manufacturers SET name = ?, country = ? WHERE id = ? AND is_deleted = FALSE;";
+    private static final String DELETE_QUERY = "UPDATE manufacturers SET is_deleted = TRUE"
+            + " WHERE id = ? AND is_deleted = FALSE;";
+
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement createStatement
+                        = connection.prepareStatement(INSERT_QUERY,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            createStatement.setString(1, manufacturer.getName());
+            createStatement.setString(2, manufacturer.getCountry());
+            createStatement.executeUpdate();
+            ResultSet generatedKeys = createStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                manufacturer.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't insert manufacturer "
+                    + manufacturer + " to DataBase", e);
+        }
+        return manufacturer;
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getStatement =
+                        connection.prepareStatement(GET_QUERY)) {
+            getStatement.setLong(1, id);
+            ResultSet resultSet = getStatement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(getManufacturersInstance(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get manufacturer with id: "
+                    + id + " from DB", e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Manufacturer> getAll() {
+        List<Manufacturer> allManufacturers = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                Statement getAllStatement = connection.createStatement()) {
```

Comment:
```suggestion
                PreparedStatement getAllStatement = connection.createStatement()) {
```
According the common mistakes